core/store/src/trie/state_parts.rs

```diff
@@ -37,7 +37,8 @@ use std::sync::Arc;
 
 /// Trie key in nibbles corresponding to the right boundary for the last state part.
 /// Guaranteed to be bigger than any existing trie key.
-const LAST_STATE_PART_BOUNDARY: &[u8; 1] = &[16];
+const LAST_STATE_PART_BOUNDARY: &[u8; 1] = &[0xFFu8];
+const LAST_STATE_PART_BOUNDARY_NIBBLES: &[u8; 2] = &[0xFu8, 0xFu8];
```

Comment:
Messed up the semantics here a bit. Had a conversation with @darioush and the issue is that originally we had `LAST_STATE_PART_BOUNDARY` as 16 which is higher than the highest possible nibble of 15. This was possible to represent in u8. However now, I need the right upper bound which is impossible to represent in `Vec<u8>`.

I'll try to shift to an `Option<Vec<u8>>` model where `None` signifies right bound. Let's see...